src/cursor/cur_table.c

```diff
@@ -617,6 +630,19 @@ __curtable_remove(WT_CURSOR *cursor)
 	}
 
 	APPLY_CG(ctable, remove);
+	WT_ERR(ret);
+
+	/*
+	 * If the cursor was positioned, it stays positioned with a key but no
+	 * no value, otherwise, there's no position, key or value. This isn't
+	 * just cosmetic, without a reset, iteration on this cursor won't start
+	 * at the beginning/end of the table.
+	 */
+	F_CLR(primary, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET);
+	if (positioned)
+		F_SET(primary, WT_CURSTD_KEY_INT);
+	else
+		APPLY_CG(ctable, reset);
```

Comment:
No, we need more. If there are indices, we do a search on each column-group cursor, which "positions" the cursor, followed by a "remove" on each column-group. The underlying remove sees the cursor is positioned, and for that reason leaves it positioned, which means that absent a reset, the next iteration will be incorrect (not to mention we're holding resources we don't want to be holding).

Theoretically the search and position should make the subsequent remove faster, because we can skip the tree search on the remove. (Although to be honest, I've never been able to make cursor remove skip the full tree search -- for some reason I haven't yet figured out, that change breaks the test suite big time.)

This is one of the reasons to add a new flag/method: table layer operations that translate into multiple underlying cursor operations set temporary cursor positions, so the underlying layer won't get the overall operation correct. (To be clear, I'm not arguing for a new flag/method, just noting this is one of the reasons that I considered adding something like it.)